app/models/user.rb

```diff
@@ -29,7 +31,35 @@ class User < ActiveRecord::Base
   validates :password, :password_confirmation,
             length: { minimum: 6, maximum: 128 }, on: :update, allow_blank: true
 
-  def following?(other_user)
-    followees.include?(other_user)
+  def self.activity_summary
+    users = User.all
```

Comment:
I think all this functionality should be done outside of User, perhaps in a query file inside a queries folder inside app,  as to not overcomplicate the User model.
Also this works great, but perhaps we could optimize this using raw queries instead of loops, the end result is not an active record anyway